tasty-plutus/src/Test/Tasty/Plutus/Internal/Minting.hs

```diff
@@ -0,0 +1,84 @@
+module Test.Tasty.Plutus.Internal.Minting (
+  -- * Types
+  MintingPolicyTask (..),
+  MintingPolicyAction (..),
+  Tokens (..),
+
+  -- * Helpers
+  burnTokens,
+  mintTokens,
+) where
+
+import Data.List.NonEmpty (NonEmpty ((:|)))
+import Plutus.V1.Ledger.Api (TokenName)
+import PlutusTx.Positive (Positive)
+import Test.QuickCheck (
+  Arbitrary (arbitrary, shrink),
+  CoArbitrary (coarbitrary),
+  Function (function),
+  functionMap,
+ )
+import Test.QuickCheck.Plutus.Instances ()
+import Prelude
+
+{- | The task for the tested minting policy.
+
+ @since 6.0
+-}
+data MintingPolicyTask
+  = MPTask MintingPolicyAction Tokens
+  deriving stock
+    ( -- | @since 6.0
+      Eq
+    , -- | @since 6.0
+      Show
+    )
+
+{- | 'MintingPolicyAction' defines a list of actions for the minting policy.
+
+ @since 6.0
+-}
+data MintingPolicyAction
+  = BurnAction
+  | MintAction
+  deriving stock (Eq, Show)
+
+{- | 'Tokens' corresponds to some positive number of 'TokenName' belonging
+ to the tested minting policy.
+
+ @since 6.0
+-}
+data Tokens = Tokens TokenName Positive
+  deriving stock (Eq, Show)
```

Comment:
Maybe the range of values represented by "Tokens" is too restrictive -
for example, you might want to deal with a bundle of various Tokens with different TokenNames from the same policy,  like:
```haskell
[("A", 1), ("B", 1)]
```
Reviewing this, it's unclear to me whether this really presents a problem for the API, as you can likely just revert to using `Value` as before, but it may be worth addressing.

Another thing is minting and burning from the same policy in the same transaction, with different token names-

```haskell
minted = 
  Value.singleton someSym "1" (-1) 
    <> Value.singleton someSym "2" 1
```

Reply:
If we need to mint/burn several Tokens, we should pass to TX `NonEmpty MintingPolicyTask` within `MintingTestItems`. In ContextBuilder we can`t use Value, because we build abstract Context Builder and don't know CurencySymbol of the tested MP.

Reply:
@t1lde - are you OK with @sergesku's explanation? I'd like to merge this, but if you think we can do better, I'd rather we did better.

Reply:
Yeah, I think that should be fine :+1: